src/Optimization/hiopAlgPrimalDecomp.cpp

```diff
@@ -1204,6 +1236,523 @@ void hiopAlgPrimalDecomposition::set_alpha_max(const double alp_max)
       return Solve_Success;    
     }
   }
+
+  /**
+   * A different asynchronous communication scheme where evaluator ranks 
+   * transfer data only once with the master rank
+   *
+   */
+  hiopSolveStatus hiopAlgPrimalDecomposition::run_local()
+  {
+    log_->printf(hovSummary, "===============\nHiop Primal Decomposition SOLVER\n===============\n");
+    log_->printf(hovSummary, "===============\nUsing local accumulation OPTION\n===============\n");
+    if(options_->GetString("print_options") == "yes") {
+      log_->write(nullptr, *options_, hovSummary);
+    }
+    if(comm_size_==1) {
+      return run_single();//call the serial solver
+    }
+    if(my_rank_==0) {
+      printf("total number of recourse problems  %lu\n", S_);
```

Comment:
`log_->printf(hovSummary, ...` instead of `printf`

check the entire file, I think there are other `printf`s

Reply:
changed to log_printf